cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -128,12 +129,13 @@ data PreBuildHooks = PreBuildHooks {
 -- | This holds the context between the pre-build, build and post-build phases.
 --
 data ProjectBuildContext = ProjectBuildContext {
-      projectRootDir   :: FilePath,
-      distDirLayout    :: DistDirLayout,
-      elaboratedPlan   :: ElaboratedInstallPlan,
-      elaboratedShared :: ElaboratedSharedConfig,
-      pkgsBuildStatus  :: BuildStatusMap,
-      buildSettings    :: BuildTimeSettings
+      projectRootDir         :: FilePath,
+      distDirLayout          :: DistDirLayout,
+      elaboratedPlanOriginal :: ElaboratedInstallPlan,
+      elaboratedPlanExecute  :: ElaboratedInstallPlan,
```

Comment:
Will add comments. The original is the improved plan before we select a subset based on targets, while the one to execute is after selecting a subset and doing the dry-run to see what's up to date.

We need the original one to get a view of the whole project, if we want to have status info that talks about the whole project.